pkg/networkservice/common/clienturl/client.go

```diff
@@ -77,11 +80,11 @@ func (u *clientURLClient) init() error {
 			return
 		}
 
-		ctx, cancel := context.WithTimeout(u.ctx, dialTimeout)
+		ctx, cancel := context.WithTimeout(u.ctx, u.dialTimeout)
 		defer cancel()
 
 		var cc *grpc.ClientConn
-		cc, u.dialErr = grpc.DialContext(ctx, grpcutils.URLToTarget(clientURL), u.dialOptions...)
+		cc, u.dialErr = grpc.DialContext(ctx, grpcutils.URLToTarget(clientURL), append(u.dialOptions, grpc.WithReturnConnectionError())...)
```

Comment:
Why do we need this WithReturnConnectionError?

Reply:
OK, this is useful https://github.com/grpc/grpc-go/blob/master/dialoptions.go#L293-L301